test/component/client.go

```diff
@@ -0,0 +1,93 @@
+package component
```

Comment:
nits: If this file is used only for testing, it may be better to change the file name to `client_test.go`.
The binary may contain this file by mistake. 

>For example, if we would have a file called person.go its test file must be named person_test.go. This is due to the package building process, where Go knows to ignore these files when building the package due to their naming. 

https://ieftimov.com/post/testing-in-go-naming-conventions/

But I couldn't find this explanation in the official page. I might be too worried. If so, never mind.

Reply:
You're right @knqyf263 This should be called `client_test.go` for the reason you've mentioned. But also to not decrease our test coverage, i.e. codecov would complain that client.go does not have any tests ;)